extensions/machine-learning-services/src/modelManagement/deployedModelService.ts

```diff
@@ -86,22 +86,23 @@ export class DeployedModelService {
 		let connection = await this.getCurrentConnection();
 		if (connection) {
 			let currentModels = await this.getDeployedModels();
-			await this._modelClient.deployModel(connection, filePath);
-			let updatedModels = await this.getDeployedModels();
-			if (details && updatedModels.length >= currentModels.length + 1) {
-				updatedModels.sort((a, b) => a.id && b.id ? a.id - b.id : 0);
-				const addedModel = updatedModels[updatedModels.length - 1];
-				addedModel.title = details.title;
-				addedModel.description = details.description;
-				addedModel.version = details.version;
-				const updatedModel = await this.updateModel(addedModel);
-				if (!updatedModel) {
-					throw Error(constants.updateModelFailedError);
-				}
+			const content = await utils.readFileInHex(filePath);
+			const fileName = details?.fileName || utils.getFileName(filePath);
+			let modelToAdd: RegisteredModel = {
+				id: 0,
+				artifactName: fileName,
+				content: content,
+				title: details?.title || fileName,
+				description: details?.description,
+				version: details?.version
+			};
+			await this._queryRunner.safeRunQuery(connection, this.getInsertModelQuery(connection.databaseName, modelToAdd));
 
-			} else {
+			let updatedModels = await this.getDeployedModels();
+			if (updatedModels.length < currentModels.length + 1) {
 				throw Error(constants.importModelFailedError);
```

Comment:
when we deploy multiple together and this fails in between - we should send back the error on which model failed. since we do one at a time that should be possible right?

Reply:
we run them all as parallels so if one fail, is shouldn't affect others. but yes I'll add more details here to be clear which model failed

---
In reply to: [406534657](https://github.com/microsoft/azuredatastudio/pull/9869#discussion_r406534657) [](ancestors = 406534657)